### PR TITLE
Start using Include FHIR server

### PR DESCRIPTION
**Pull request by celinepelletier**
Add Include fhir server config
Add some new fields
Refactor to make ETL compatible with Include and KF Fhir 
Aggregate participant biospecimen and file

### CODE REVIEW

File: import-task/src/main/scala/bio/ferlab/fhir/etl/Utils.scala [hunk 1]
```diff
@@ -23,6 +24,8 @@ object Utils {
   val extractAclFromList: UserDefinedFunction =
     udf((arr: Seq[String]) => arr.filter(e => (e matches actCodeR) || (e matches studyCodePattern)))
 
+  val extractReferencesId: UserDefinedFunction = udf((arr: Seq[String]) => arr.map(e  => e.split("/")(1)))
```
Comment (botekchristophe) [inline import-task/src/main/scala/bio/ferlab/fhir/etl/Utils.scala:27]: you can avoid using a udf by using org.apache.spark.sql.functions.transform()
```
import org.apache.spark.sql.functions

functions.transform(column, c => functions.split(c, "/")(1))
```

**Review by botekchristophe — APPROVED** [review]

File: import-task/src/main/scala/bio/ferlab/fhir/etl/Utils.scala (hunk 1 above)
Comment (jecos) [reply import-task/src/main/scala/bio/ferlab/fhir/etl/Utils.scala:27]: +1

File: import-task/src/main/scala/bio/ferlab/fhir/etl/transformations/Transformations.scala [hunk 2]
```diff
@@ -10,34 +10,30 @@ import org.apache.spark.sql.types.BooleanType
...
   )
 
   val researchSubjectMappings: List[Transformation] = List(
     Custom(_
       .select("fhir_id", "release_id", "identifier")
-      .withColumn("external_id", col("identifier")(0)("value"))
-      .withColumn("participant_id", regexp_extract(col("identifier")(2)("value"), patternParticipantStudy, 2))
-      .withColumn("study_id", regexp_extract(col("identifier")(2)("value"), patternParticipantStudy, 1))
+      .withColumn("external_id", filter(col("identifier"), c => c("system").isNull)(0)("value"))
+      .withColumn("participant_id", filter(col("identifier"), c => c("system") === s"${SYS_DATASERVICE_URL}participants/")(0)("value"))
```
Comment (jecos) [inline import-task/src/main/scala/bio/ferlab/fhir/etl/transformations/Transformations.scala:35]: it worth it to have a function to return the first element of a coding that match a system

File: prepare-index/src/main/scala/bio/ferlab/fhir/etl/centricTypes/FileCentric.scala [hunk 3]
```diff
@@ -38,25 +43,25 @@ class FileCentric(releaseId: String, studyIds: List[String])(implicit configurat
...
-    Map("es_index_file_centric" -> transformedFile)
+    Map(mainDestination.id -> transformedFile)
   }
 
   override def load(data: Map[String, DataFrame],
                     lastRunDateTime: LocalDateTime = minDateTime,
                     currentRunDateTime: LocalDateTime = LocalDateTime.now())(implicit spark: SparkSession): Map[String, DataFrame] = {
-    println(s"COUNT: ${data("es_index_file_centric").count()}")
-    val dataToLoad = Map("es_index_file_centric" -> data("es_index_file_centric")
+    println(s"COUNT: ${data(mainDestination.id).count()}")
```
Comment (jecos) [inline prepare-index/src/main/scala/bio/ferlab/fhir/etl/centricTypes/FileCentric.scala:63]: remove the count.

File: prepare-index/src/main/scala/bio/ferlab/fhir/etl/centricTypes/ParticipantCentric.scala [hunk 4]
```diff
@@ -37,23 +42,23 @@ class ParticipantCentric(releaseId: String, studyIds: List[String])(implicit con
...
-    Map("es_index_participant_centric" -> transformedParticipant)
+    Map(mainDestination.id -> transformedParticipant)
   }
 
   override def load(data: Map[String, DataFrame],
                     lastRunDateTime: LocalDateTime = minDateTime,
                     currentRunDateTime: LocalDateTime = LocalDateTime.now())(implicit spark: SparkSession): Map[String, DataFrame] = {
-    println(s"COUNT: ${data("es_index_participant_centric").count()}")
-    val dataToLoad = Map("es_index_participant_centric" -> data("es_index_participant_centric")
+    println(s"COUNT: ${data(mainDestination.id).count()}")
```
Comment (jecos) [inline prepare-index/src/main/scala/bio/ferlab/fhir/etl/centricTypes/ParticipantCentric.scala:60]: remove the count

File: prepare-index/src/main/scala/bio/ferlab/fhir/etl/centricTypes/SimpleParticipant.scala [hunk 5]
```diff
@@ -69,14 +69,14 @@ class SimpleParticipant(releaseId: String, studyIds: List[String])(implicit conf
...
-    Map("simple_participant" -> transformedParticipant)
+    Map(mainDestination.id -> transformedParticipant)
   }
 
   override def load(data: Map[String, DataFrame],
                     lastRunDateTime: LocalDateTime = minDateTime,
                     currentRunDateTime: LocalDateTime = LocalDateTime.now())(implicit spark: SparkSession): Map[String, DataFrame] = {
-    println(s"COUNT: ${data("simple_participant").count()}")
-    val dataToLoad = Map("simple_participant" -> data("simple_participant")
+    println(s"COUNT: ${data(mainDestination.id).count()}")
```
Comment (jecos) [inline prepare-index/src/main/scala/bio/ferlab/fhir/etl/centricTypes/SimpleParticipant.scala:78]: remove the count

**Review by jecos — CHANGES_REQUESTED** [review]

**Review by adipaul1981 — APPROVED** [review]

**Review by jecos — APPROVED** [review]

**Pull request merged**